software/controller/lib/hal/gpio.h

```diff
@@ -0,0 +1,101 @@
+/* Copyright 2020-2021, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This file defines GPIO functions for HAL (Hardware Abstraction Layer) for the
+STM32L452 processor used on the controller.
+
+Reference abbreviations [RM], [DS], etc are defined in hal/README.md.
+*/
+
+#pragma once
+
+#include <cstdint>
+
+// General Purpose I/O [RM] 8.4
+
+namespace GPIO {
+
+/// \TODO: Move these mappings elsewhere
+enum class Port : uint32_t {
+  PortA = 0x48000000,
```

Comment:
IMO, `GPIO::Port::A` rolls of better than `GPIO::Port::PortA`, though I agree it's better to have the enum names themselves be explicit (especially when `using` exists).

Reply:
Well, `using` should be used very sparingly anyhow, so actually I think you have a point there. In fact, if it was less wordy, there would be less of a reason for `using`.